File: application/src/main/java/de/pbauerochse/worklogviewer/youtrack/v20174/ReportStatus.java
package de.pbauerochse.worklogviewer.youtrack.v20174;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

@JsonIgnoreProperties(ignoreUnknown = true)
public class ReportStatus {

    private final boolean calculationInProgress;
    private final int progress;

    @JsonCreator
    public ReportStatus(@JsonProperty("calculationInProgress") boolean calculationInProgress,
                        @JsonProperty("progress") int progress) {
        this.calculationInProgress = calculationInProgress;
        this.progress = progress;
    }

    public boolean isCalculationInProgress() {
        return calculationInProgress;
    }

    public int getProgress() {
        return progress;
    }
}

File: connectors/v20174_to_20181/src/main/java/de/pbauerochse/worklogviewer/connector/v2017/domain/report/ReportStatus.kt
package de.pbauerochse.worklogviewer.connector.v2017.domain.report

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
class ReportStatus @JsonCreator constructor(
    @JsonProperty("calculationInProgress") val calculating : Boolean,
    @JsonProperty("progress") val progress : Int
)